File: articles/object-graphs-3/dave2.js
// Make a parent class
function Person() {}
// with an instance method
Person.prototype.greet = function greet() {
  console.log("Hello");
}
// and a class method.
Person.create = function create() {
  return new this();
};

// Create a subclass
function Dave() {}
Dave.__proto__ = Person;
Dave.prototype.__proto__ = Person.prototype;
// and test it.
console.log(Dave.create());
console.log(new Dave);
Dave.create().greet();

File: articles/object-graphs-3/dave2.rb
# Make a parent class
class Person
  # with an instance method
  def greet
    puts "Hello"
  end
  # and a class method.
  def self.create
    self.new
  end
end

# Create a subclass
class Dave < Person
end
# and test it.
puts Dave.create
puts Dave.new
Dave.create.greet
